src/orange/detail/BIHData.hh

```diff
@@ -24,37 +24,43 @@ namespace detail
 /*!
  * Data for a single inner node in a Bounding Interval Hierarchy.
  *
- * Note that the LEFT bounding plane position is the far right boundary of the
- * left side of the tree, and the RIGHT bounding plane position is the far left
- * boundary of the right side of the tree. Since the halfspaces of created by
- * the bounding planes may overlap, the LEFT bounding plane position could be
- * either left or right of the RIGHT bounding plane position.
+ * As a convention, a node's LEFT edge corresponds to the half space that is
+ * less than the partition value. In other words, the LEFT bounding plane
+ * position is the far right boundary of the left side of the tree, and the
+ * RIGHT bounding plane position is the far left boundary of the right side of
+ * the tree. Since the halfspaces created by the bounding planes may overlap,
+ * the LEFT bounding plane position could be either left or right of the RIGHT
+ * bounding plane position.
  */
 struct BIHInnerNode
 {
     using real_type = fast_real_type;
 
-    struct BoundingPlane
+    struct Edge
     {
-        real_type position;
+        //! The position of the bounding plane along the partition axis
+        real_type bounding_plane_pos;
+        //! The child node connected to this edge
         BIHNodeId child;
+        //! Bbox created by clipping an inf bbox with the bounding planes
+        //! between this edge (inclusive) and the root.
+        FastBBox bbox;
     };
 
-    enum class Edge
+    enum class Side
```

Comment:
You could also use `enum class Bound` (geocel/Types) lo/hi instead of this enum... might be more generic than left/right which imply x axis?

Reply:
These are in reference to the left/right edges of the tree itself, not what side of the bounding plane the point is on. Using `Bound` still makes sense semantically as the left side of the tree is the side corresponding to the "low" side of the bounding plane. However, since all of the comments in this class refer to the left/right edges (consistent with tree terminology used in the literature), I think there is value in keeping this enum for readability.